### PR TITLE
Guus' Daily Fixes

### PR DESCRIPTION
**Pull request by guusw**

### CODE REVIEW

File: src/core/runtime.cpp [hunk 1]
```diff
@@ -864,7 +864,7 @@ bool matchTypes(const SHTypeInfo &inputType, const SHTypeInfo &receiverType, boo
         }
         return false;
       } else if (inputType.seqTypes.len == 0 || receiverType.seqTypes.len == 0) {
-        return false;
```
Comment (sinkingsugar) [inline src/core/runtime.cpp:867]: bad feeling about this :)

Comment (sinkingsugar) [reply src/core/runtime.cpp:867]: but hmm makes sense to be true I guess...

Comment (Kryptos-FR) [reply src/core/runtime.cpp:867]: I think it is consistent with algorithms where `seq.any(/* whatever condition */)` returns `true` for empty sequences.

Comment (sinkingsugar) [reply src/core/runtime.cpp:867]: @guusw this is wrong, RECEIVER SHOULD have a type. no types is an anomaly, should at least have ANY. Please restore `false`

Comment (sinkingsugar) [reply src/core/runtime.cpp:867]: you can workaround `expect` issues by forcing things to be any with generic `ExpectTable` and `ExpectSeq` but let's avoid breaking delicate behavior

Comment (guusw) [reply src/core/runtime.cpp:867]: Yeah didn't realize what this was meant for, will add a comment. undo and work around it for now

**Review by sinkingsugar — APPROVED** [review]

**Pull request merged**